<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="Visualizing and interpreting Numbat outputs"><title>Interpreting Numbat results â€¢ numbat</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Interpreting Numbat results"><meta property="og:description" content="Visualizing and interpreting Numbat outputs"><meta property="og:image" content="https://kharchenkolab.github.io/numbat/logo.png"><meta name="google-site-verification" content="GE78cdBqx5YPEbHjqtohe_p51Ng8oEPXdcaSEIu2dP8"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">numbat</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.2</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="../articles/numbat.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/results.html">Interpreting Numbat results</a>
    <a class="dropdown-item" href="../articles/hmm.html">Numbat haplotype-aware HMM</a>
  </div>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/kharchenkolab/numbat/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-article">

<div class="row">
  <main id="main"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Interpreting Numbat results</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/kharchenkolab/numbat/blob/HEAD/vignettes/results.Rmd" class="external-link"><code>vignettes/results.Rmd</code></a></small>
      <div class="d-none name"><code>results.Rmd</code></div>
    </div>

    
    
<p>In this tutorial, we will illustrate how to visualize and interpret Numbat output, using a triple-negative breast cancer dataset (TNBC1) from <a href="https://www.nature.com/articles/s41587-020-00795-2">Gao et al</a>. We will use <a href="https://github.com/kharchenkolab/pagoda2">pagoda2</a> for visualizing cells in low-dimensional expression space.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(ggplot2)
<span class="kw">library</span>(numbat)
<span class="kw">library</span>(dplyr)
<span class="kw">library</span>(glue)
<span class="kw">library</span>(data.table)
<span class="kw">library</span>(ggtree)
<span class="kw">library</span>(stringr)
<span class="kw">library</span>(tidygraph)
<span class="kw">library</span>(patchwork)</code></pre></div>
<p>After a Numbat run, we can summarize output files into a Numbat object:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">nb =<span class="st"> </span>Numbat<span class="op">$</span><span class="kw">new</span>(<span class="dt">out_dir =</span> <span class="st">&#39;./numbat_out&#39;</span>)</code></pre></div>
<p>In this tutorial, let's start the analysis with the pre-saved Numbat and pagoda2 objects.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">nb =<span class="st"> </span><span class="kw">readRDS</span>(<span class="kw">url</span>(<span class="st">&#39;http://pklab.med.harvard.edu/teng/data/nb_TNBC1.rds&#39;</span>))
pagoda =<span class="st"> </span><span class="kw">readRDS</span>(<span class="kw">url</span>(<span class="st">&#39;http://pklab.med.harvard.edu/teng/data/con_TNBC1.rds&#39;</span>))</code></pre></div>
<div id="copy-number-landscape-and-single-cell-phylogeny" class="section level2">
<h2>Copy number landscape and single-cell phylogeny</h2>
<p>As an overview, we can visualize the CNV calls in single-cells and their evolutionary relationships in an integrated plot panel:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">mypal =<span class="st"> </span><span class="kw">c</span>(<span class="st">&#39;1&#39;</span> =<span class="st"> &#39;gray&#39;</span>, <span class="st">&#39;2&#39;</span> =<span class="st"> &quot;#377EB8&quot;</span>, <span class="st">&#39;3&#39;</span> =<span class="st"> &quot;#4DAF4A&quot;</span>, <span class="st">&#39;4&#39;</span> =<span class="st"> &quot;#984EA3&quot;</span>)
  
nb<span class="op">$</span><span class="kw">plot_phylo_heatmap</span>(
  <span class="dt">clone_bar =</span> <span class="ot">TRUE</span>, 
  <span class="dt">p_min =</span> <span class="fl">0.9</span>,
  <span class="dt">pal_clone =</span> mypal
)</code></pre></div>
<p><img src="results_files/figure-html/unnamed-chunk-5-1.png" width="2400" /> In this visualization, the single-cell phylogeny (left) is juxtaposed with a heatmap of single-cell CNV calls (right). The CNV calls are colored by the type of alterations (AMP, amplification, BAMP, balanced amplification, DEL, deletion, CNLoH, copy-neutral loss of heterozygosity). The colorbar in-between differentiates the distinct genetic populations (genotype). The dashed blue line separates the predicted tumor versus normal cells. This tells us that the dataset mainly consists of three cell populations, a normal population (gray) and two tumor subclones (green and yellow).</p>
</div>
<div id="consensus-copy-number-segments" class="section level2">
<h2>Consensus copy number segments</h2>
<p>Let's take a look at the consensus segments.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">nb<span class="op">$</span><span class="kw">plot_consensus</span>()</code></pre></div>
<p><img src="results_files/figure-html/unnamed-chunk-6-1.png" width="4200" /></p>
</div>
<div id="bulk-cnv-profiles" class="section level2">
<h2>Bulk CNV profiles</h2>
<p>We can also visualize these CNV events in pseudobulks where the data is more rich, aggregating cells by clone:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">nb<span class="op">$</span>bulk_clones <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span><span class="kw">filter</span>(n_cells <span class="op">&gt;</span><span class="st"> </span><span class="dv">50</span>) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">plot_bulks</span>(
    <span class="dt">min_LLR =</span> <span class="dv">10</span>, <span class="co"># filtering CNVs by evidence</span>
    <span class="dt">legend =</span> <span class="ot">FALSE</span>
  )</code></pre></div>
<p><img src="results_files/figure-html/unnamed-chunk-7-1.png" width="3600" /></p>
</div>
<div id="single-cell-cnv-calls" class="section level2">
<h2>Single-cell CNV calls</h2>
<p>Numbat probabilistically evaluates the presence/absence of CNVs in single cells. The cell-level CNV posteriors are stored in the <code>nb$joint_post</code> dataframe:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">head</span>(nb<span class="op">$</span>joint_post) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">select</span>(cell, CHROM, seg, cnv_state, p_cnv, p_cnv_x, p_cnv_y)</code></pre></div>
<pre><code>##                        cell CHROM seg cnv_state        p_cnv     p_cnv_x
## 1: TNBC1_AAACCTGCACCTTGTC-1     1  1b       amp 0.9999999997 1.000000000
## 2: TNBC1_AAACCTGCACCTTGTC-1     2  2a       amp 0.0002288397 0.001172894
## 3: TNBC1_AAACCTGCACCTTGTC-1     3  3a       amp 0.0522751360 0.052273344
## 4: TNBC1_AAACCTGCACCTTGTC-1     3  3c       amp 0.9147696619 0.982278728
## 5: TNBC1_AAACCTGCACCTTGTC-1     4  4b       amp 0.8999658266 0.845619643
## 6: TNBC1_AAACCTGCACCTTGTC-1     5  5a       amp 0.8070027728 0.778074329
##       p_cnv_y
## 1: 0.03844235
## 2: 0.09821435
## 3: 0.50000000
## 4: 0.10246769
## 5: 0.61862288
## 6: 0.53344303</code></pre>
<p>which contains cell-level information on specific CNV segments (<code>seg</code>), their alteration type (<code>cnv_state</code>), the joint posterior probability of the CNV (<code>p_cnv</code>), the expression-based posterior (<code>p_cnv_x</code>), and the allele-based posterior (<code>p_cnv_y</code>). We can visualize the event-specific posteriors in a expression-based tSNE embedding:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">plist =<span class="st"> </span><span class="kw">list</span>()
muts =<span class="st"> </span><span class="kw">c</span>(<span class="st">&#39;1a&#39;</span>, <span class="st">&#39;3b&#39;</span>, <span class="st">&#39;22b&#39;</span>)
cnv_type =<span class="st"> </span>nb<span class="op">$</span>joint_post <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">distinct</span>(seg, cnv_state) <span class="op">%&gt;%</span><span class="st"> </span>{<span class="kw">setNames</span>(.<span class="op">$</span>cnv_state, .<span class="op">$</span>seg)}
<span class="cf">for</span> (mut <span class="cf">in</span> muts) {
    
    plist[[mut]] =<span class="st"> </span>pagoda<span class="op">$</span><span class="kw">plotEmbedding</span>(
        <span class="dt">alpha=</span><span class="fl">0.8</span>,
        <span class="dt">size=</span><span class="dv">1</span>, 
        <span class="dt">plot.na =</span> F, 
        <span class="dt">colors =</span> nb<span class="op">$</span>joint_post <span class="op">%&gt;%</span>
<span class="st">            </span><span class="kw">filter</span>(seg <span class="op">==</span><span class="st"> </span>mut) <span class="op">%&gt;%</span>
<span class="st">            </span>{<span class="kw">setNames</span>(.<span class="op">$</span>p_cnv, .<span class="op">$</span>cell)},
        <span class="dt">show.legend =</span> T,
        <span class="dt">mark.groups =</span> F,
        <span class="dt">plot.theme =</span> <span class="kw">theme_bw</span>(),
        <span class="dt">title =</span> <span class="kw">paste0</span>(mut, <span class="st">&#39;(&#39;</span>, cnv_type[muts], <span class="st">&#39;)&#39;</span>)
    ) <span class="op">+</span>
<span class="st">    </span><span class="kw">scale_color_gradient2</span>(<span class="dt">low =</span> <span class="st">&#39;royalblue&#39;</span>, <span class="dt">mid =</span> <span class="st">&#39;white&#39;</span>, <span class="dt">high =</span> <span class="st">&#39;red3&#39;</span>, <span class="dt">midpoint =</span> <span class="fl">0.5</span>, <span class="dt">limits =</span> <span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">1</span>), <span class="dt">name =</span> <span class="st">&#39;Posterior&#39;</span>)
}
<span class="kw">wrap_plots</span>(plist, <span class="dt">guides =</span> <span class="st">&#39;collect&#39;</span>)</code></pre></div>
<p><img src="results_files/figure-html/unnamed-chunk-9-1.png" width="2700" /></p>
</div>
<div id="clonal-assignments" class="section level2">
<h2>Clonal assignments</h2>
<p>Numbat aggregates signals across subclone-specific CNVs to probabilistically assign cells to subclones. The information regarding clonal assignments are contained in the <code>nb$clone_post</code> dataframe.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">nb<span class="op">$</span>clone_post <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">head</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">select</span>(cell, clone_opt, p_<span class="dv">1</span>, p_<span class="dv">2</span>, p_<span class="dv">3</span>, p_<span class="dv">4</span>)</code></pre></div>
<pre><code>##                        cell clone_opt           p_1          p_2          p_3
## 1: TNBC1_AAACCTGCACCTTGTC-1         4  1.162495e-72 2.105425e-20 3.478936e-29
## 2: TNBC1_AAACGGGAGTCCTCCT-1         1  1.000000e+00 5.814741e-51 5.920463e-95
## 3: TNBC1_AAACGGGTCCAGAGGA-1         4 2.434561e-100 2.175022e-14 2.085718e-30
## 4: TNBC1_AAAGATGCAGTTTACG-1         3  2.020088e-23 8.164920e-04 9.913458e-01
## 5: TNBC1_AAAGCAACAGGAATGC-1         4  9.546322e-64 4.989037e-11 1.657378e-31
## 6: TNBC1_AAAGCAATCGGAATCT-1         3  1.516589e-99 8.466806e-33 1.000000e+00
##             p_4
## 1: 1.000000e+00
## 2: 8.821035e-66
## 3: 1.000000e+00
## 4: 7.837745e-03
## 5: 1.000000e+00
## 6: 2.457592e-29</code></pre>
<p>Here <code>clone_opt</code> denotes the maximum likelihood assignment of a cell to a specific clone. <code>p_{1..4}</code> are the detailed breakdown of the posterior probability that the cell belongs to each clone, respectively. Let's visualize the clonal decomposition in a tSNE embedding. Note that clone 1 is always the normal cells.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">pagoda<span class="op">$</span><span class="kw">plotEmbedding</span>(
    <span class="dt">alpha=</span><span class="fl">0.8</span>,
    <span class="dt">size=</span><span class="dv">1</span>, 
    <span class="dt">groups =</span> nb<span class="op">$</span>clone_post <span class="op">%&gt;%</span>
<span class="st">        </span>{<span class="kw">setNames</span>(.<span class="op">$</span>clone_opt, .<span class="op">$</span>cell)},
    <span class="dt">plot.na =</span> F,
    <span class="dt">plot.theme =</span> <span class="kw">theme_bw</span>(),
    <span class="dt">title =</span> <span class="st">&#39;Genotypes&#39;</span>,
    <span class="dt">pal =</span> mypal
)</code></pre></div>
<p><img src="results_files/figure-html/unnamed-chunk-11-1.png" width="900" /></p>
</div>
<div id="tumor-versus-normal-probability" class="section level2">
<h2>Tumor versus normal probability</h2>
<p>Combining evidence from all CNVs, Numbat derives an aneuploidy probability for each cell to distinguish tumor versus normal cells. We can visualize the posterior aneuploidy probability based on expression evidence only, allele evidence only, and jointly:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">p_joint =<span class="st"> </span>pagoda<span class="op">$</span><span class="kw">plotEmbedding</span>(
    <span class="dt">alpha=</span><span class="fl">0.8</span>,
    <span class="dt">size=</span><span class="dv">1</span>, 
    <span class="dt">colors =</span> nb<span class="op">$</span>clone_post <span class="op">%&gt;%</span>
<span class="st">        </span>{<span class="kw">setNames</span>(.<span class="op">$</span>p_cnv, .<span class="op">$</span>cell)},
    <span class="dt">plot.na =</span> F,
    <span class="dt">plot.theme =</span> <span class="kw">theme_bw</span>(),
    <span class="dt">title =</span> <span class="st">&#39;Joint&#39;</span>,
) <span class="op">+</span>
<span class="kw">scale_color_gradient2</span>(<span class="dt">low =</span> <span class="st">&#39;royalblue&#39;</span>, <span class="dt">mid =</span> <span class="st">&#39;white&#39;</span>, <span class="dt">high =</span> <span class="st">&#39;red3&#39;</span>, <span class="dt">midpoint =</span> <span class="fl">0.5</span>, <span class="dt">name =</span> <span class="st">&#39;Posterior&#39;</span>)
p_allele =<span class="st"> </span>pagoda<span class="op">$</span><span class="kw">plotEmbedding</span>(
    <span class="dt">alpha=</span><span class="fl">0.8</span>,
    <span class="dt">size=</span><span class="dv">1</span>, 
    <span class="dt">colors =</span> nb<span class="op">$</span>clone_post <span class="op">%&gt;%</span>
<span class="st">        </span>{<span class="kw">setNames</span>(.<span class="op">$</span>p_cnv_x, .<span class="op">$</span>cell)},
    <span class="dt">plot.na =</span> F,
    <span class="dt">plot.theme =</span> <span class="kw">theme_bw</span>(),
    <span class="dt">title =</span> <span class="st">&#39;Expression&#39;</span>,
) <span class="op">+</span>
<span class="kw">scale_color_gradient2</span>(<span class="dt">low =</span> <span class="st">&#39;royalblue&#39;</span>, <span class="dt">mid =</span> <span class="st">&#39;white&#39;</span>, <span class="dt">high =</span> <span class="st">&#39;red3&#39;</span>, <span class="dt">midpoint =</span> <span class="fl">0.5</span>, <span class="dt">name =</span> <span class="st">&#39;Posterior&#39;</span>)
p_expr =<span class="st"> </span>pagoda<span class="op">$</span><span class="kw">plotEmbedding</span>(
    <span class="dt">alpha=</span><span class="fl">0.8</span>,
    <span class="dt">size=</span><span class="dv">1</span>, 
    <span class="dt">colors =</span> nb<span class="op">$</span>clone_post <span class="op">%&gt;%</span>
<span class="st">        </span>{<span class="kw">setNames</span>(.<span class="op">$</span>p_cnv_y, .<span class="op">$</span>cell)},
    <span class="dt">plot.na =</span> F,
    <span class="dt">show.legend =</span> T,
    <span class="dt">plot.theme =</span> <span class="kw">theme_bw</span>(),
    <span class="dt">title =</span> <span class="st">&#39;Allele&#39;</span>,
) <span class="op">+</span>
<span class="kw">scale_color_gradient2</span>(<span class="dt">low =</span> <span class="st">&#39;royalblue&#39;</span>, <span class="dt">mid =</span> <span class="st">&#39;white&#39;</span>, <span class="dt">high =</span> <span class="st">&#39;red3&#39;</span>, <span class="dt">midpoint =</span> <span class="fl">0.5</span>, <span class="dt">name =</span> <span class="st">&#39;Posterior&#39;</span>)
(p_expr <span class="op">|</span><span class="st"> </span>p_allele <span class="op">|</span><span class="st"> </span>p_joint) <span class="op">+</span><span class="st"> </span><span class="kw">plot_layout</span>(<span class="dt">guides =</span> <span class="st">&#39;collect&#39;</span>)</code></pre></div>
<p><img src="results_files/figure-html/unnamed-chunk-12-1.png" width="2700" /> Both expression and allele signal clearly separate the tumor and normal cells.</p>
</div>
<div id="tumor-phylogeny" class="section level2">
<h2>Tumor phylogeny</h2>
<p>Let's take a closer look at the inferred single cell phylogeny and where the mutations occurred.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">nb<span class="op">$</span><span class="kw">plot_sc_tree</span>(
  <span class="dt">label_size =</span> <span class="dv">3</span>, 
  <span class="dt">branch_width =</span> <span class="fl">0.5</span>, 
  <span class="dt">tip_length =</span> <span class="fl">0.5</span>,
  <span class="dt">pal_clone =</span> mypal, 
  <span class="dt">tip =</span> <span class="ot">TRUE</span>
)</code></pre></div>
<p><img src="results_files/figure-html/unnamed-chunk-13-1.png" width="1800" /> The mutational history can also be represented on the clone level, where cells with the same genotype are aggregated into one node.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">nb<span class="op">$</span><span class="kw">plot_mut_history</span>(<span class="dt">pal =</span> mypal)</code></pre></div>
<p><img src="results_files/figure-html/unnamed-chunk-14-1.png" width="1800" /></p>
</div>
  </main>
   <aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
  
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Teng Gao, Hirak Sarkar, Evan Biederstedt, Peter Kharchenko.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.3.</p>
</div>

    </footer></div>

  

  

  </body></html>
